### PR TITLE
Fix form widget aliases

### LINKED ISSUE #419: The title field is required error appears sometimes even if I already have a title
If I create a page and save it then I want to add another one, I get the following error even if I have a title that is required:

image

Basically, I have to refresh the backend everytime I want to add a new page. 

The same issue appears if I enter the title then url then select a layout but this time the URL is marked as "required" - then I have to refresh again and select "Layout" first.

### PR DESCRIPTION
**Pull request by bennothommo**
- Create a unique alias if the object is new - ie. new page or menu
- Ensure widget alias is kept in between refreshes and AJAX requests

Fixes #419 and https://github.com/octobercms/october/issues/4628.

### CODE REVIEW

**Comment by LukeTowers** [comment]
@bennothommo how does this tie in with https://github.com/octobercms/october/commit/c23d671f9104f2239fc77c9601d48eb5c6db5c3a? Do we need to make sure that whatever the final solution is here we apply it to both?

**Comment by bennothommo** [comment]
@LukeTowers it may need the same fix in CMS as well, but I want to see if this fix is effective first before rolling it out there.

File: controllers/Index.php [hunk 1]
```diff
@@ -509,7 +509,11 @@ protected function makeObjectFormWidget($type, $object, $alias = null)
 
         $widgetConfig = $this->makeConfig($formConfigs[$type]);
         $widgetConfig->model = $object;
-        $widgetConfig->alias = $alias ?: 'form'.studly_case($type).md5($object->getFileName());
+        if (!$object->exists) {
+            $widgetConfig->alias = $alias ?: 'form' . studly_case($type) . uniqid();
+        } else {
+            $widgetConfig->alias = $alias ?: 'form' . studly_case($type) . md5($object->getFileName());
+        }
```
Comment (LukeTowers) [inline controllers/Index.php]: Can you simplify this to match https://github.com/octobercms/october/commit/c23d671f9104f2239fc77c9601d48eb5c6db5c3a?

Comment (bennothommo) [reply controllers/Index.php]: But that one-liner makes my eye twitch :P

Comment (LukeTowers) [reply controllers/Index.php]: I know, but it's how it was done in the main repo so I'd prefer it to match. And relatively speaking it's not that bad 😝

Comment (bennothommo) [reply controllers/Index.php]: @LukeTowers sorted. (*eyes twitch further*)

Comment (LukeTowers) [reply controllers/Index.php]: You should uh, get that looked at 😆

**Comment by LukeTowers** [comment]
@bennothommo just using https://github.com/rainlab/pages-plugin/commit/60a22613cbc88ce0c750bb634a8371c7ae83ce4b alone seems to fix the issue for me, although I would think the passing the alias would be required. Could you double check to see if it's needed or not?

**Comment by bennothommo** [comment]
@LukeTowers I'm fairly certain it was needed. I had initially tried just the alias fix that @daftspunk ended up implementing, but I still encountered the problems that were reported in the linked issues.

**Comment by LukeTowers** [comment]
@bennothommo could you test it out again with the latest code base? I was testing it with a repeater and a mediafinder and wasn't having any problems adding media files to newly created pages or items in the repeater.

**Comment by LukeTowers** [comment]
@bennothommo I've tested it out myself with the media manager onUpload changes and can verify that it's required. Thanks for your work on this!

**Pull request merged**